### ISSUE TITLE
Unexpected error 'Second argument to "super" call must be object or class that derives from XX'

### ISSUE THREAD

**Issue by lukas-lang** [microsoft/pyright#10388]
## Environment data

-   Pylance version: 2025.4.1
-   OS and version: Windows 11, 24H2
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.9

## Code Snippet

```python
from typing import Type

class Base: ...

class Sub(Base): ...

def foo(cls: Type[Base]):
    if True:
        super(Base, cls)
    if issubclass(cls, Sub):
        super(Base, cls)
    if issubclass(cls, Sub) or True:
        super(Base, cls)

foo(Base)
foo(Sub)
```

## Repro Steps

Create a file with the above code.

## Expected behavior

Pylance should recognize that the call is actually valid, as it does properly in the the two previous cases.

## Actual behavior

The third call to `super` causes the warning
```
Second argument to "super" call must be object or class that derives from "type[Base]"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
(parameter) cls: type[Sub] | type[Base]
```

It looks like somehow the union type deduced for `cls` confuses the type checker - the following also produces the same warning in the third case:

```python
def bar1(cls: Type[Base]):
    super(Base, cls)

def bar2(cls: Type[Sub]):
    super(Base, cls)

def bar3(cls: Type[Base] | Type[Sub]):
    super(Base, cls)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.401.
